ring_theory/noetherian.lean

```diff
@@ -89,6 +89,26 @@ theorem is_noetherian_iff_well_founded
 
 def is_noetherian_ring (α) [ring α] : Prop := is_noetherian α α
 
+theorem ring.is_noetherian_of_fintype (R M) [ring R] [module R M] [fintype M] : is_noetherian R M :=
+begin
+  intro N,
+  existsi (to_finset N : finset M),
+  suffices : span (N : set M) = N,
+    simpa,
+  exact span_eq_of_is_submodule N.to_is_submodule,
+end
+
+noncomputable instance fintype.of_subsingleton_ring {α} [ring α] [h : subsingleton α] : fintype α :=
```

Comment:
I'm surprised that this is `noncomputable`. The computational part seams to be completely fine...